### PR TITLE
crimson: add crimson-osd rpm and deb runtime dependencies for protobuf inherited from seastar

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -392,6 +392,7 @@ Depends: ceph-base (= ${binary:Version}),
          ${misc:Depends},
          ${python3:Depends},
          ${shlibs:Depends},
+         libprotobuf23 <pkg.ceph.crimson>,
```
Comment (athanatos) [inline debian/control:395]: libprotobuf23 seems to be the right package from jammy -- I don't see a way to do this more generally?

**Review by dmick — APPROVED** [review]
This should be close to what we need, especially when Ken's fix to the el8 copr goes through

**Comment by dmick** [comment]
jenkins retest

**Comment by dmick** [comment]
jenkins test api

**Comment by dmick** [comment]
jenkins test make check

**Comment by dmick** [comment]
bypassing make check because Jenkins is *slammed* and this is holding up other testing

**Comment by dmick** [comment]
Looks as though this needs backport to other branches as well

**Comment by Svelar** [comment]
> Looks as though this needs backport to other branches as well

Yes.

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/77881/consoleFull#17838507786030192d-da7d-4544-bbc5-ba785a77a7b9:~:text=Error%3A%20%0A%20Problem%201,best%20candidate%20packages)

**Comment by dmick** [comment]
Filed https://tracker.ceph.com/issues/65123 to mechanize backport(s)

File: ceph.spec.in [hunk 2]
```diff
@@ -887,6 +887,9 @@ Provides:	ceph-test:/usr/bin/ceph-osdomap-tool
 Requires:	ceph-base = %{_epoch_prefix}%{version}-%{release}
 Requires:	sudo
 Requires:	libstoragemgmt
+%if 0%{with seastar}
+Requires:	protobuf
+%endif
```
Comment (ktdreyer) [inline ceph.spec.in:892]: `rpmbuild` will automatically add a `libprotobuf.so.25()(64bit)` dependency to the packages that need it (eg `ceph-osd`), so we don't technically need this `Requires` line in `ceph.spec.in`. It's harmless, but we could remove it.

**Pull request merged**